### PR TITLE
Make Bond Issuance work on DABL

### PR DESCRIPTION
**Pull request by gaborh-da**

### CODE REVIEW

File: src/main/daml/DA/RefApps/Bond/Test/MarketSetupScript.daml [hunk 1]
```diff
@@ -31,7 +31,23 @@ setupMarket = do
...
+    bank1 : Party
+    bank2 : Party
+    bank3 : Party
+    issuer : Party
+    centralBank : Party
+    regulator : Party
+    operator : Party
+
+setupMarketWithParties : BondIssuanceParties -> Script ()
+setupMarketWithParties parties@BondIssuanceParties{..} = do
```
Comment (tamaskalcza-da) [inline src/main/daml/DA/RefApps/Bond/Test/MarketSetupScript.daml:49]: Wouldn't this be the same?
```suggestion
setupMarketWithParties parties = do
```

File: triggers/src/main/daml/DA/RefApps/Bond/Triggers/TriggersLibrary.daml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import Daml.Trigger
+
+import DA.RefApps.Bond.Triggers.RedemptionFinalizeTrigger
+import DA.RefApps.Bond.Triggers.AuctionFinalizeTrigger
+import DA.RefApps.Bond.Triggers.RedemptionCalculationTrigger
+import DA.RefApps.Bond.Triggers.PlaceBidTrigger
+import DA.RefApps.Bond.Triggers.InvestorSettlementTrigger
+import DA.RefApps.Bond.Triggers.CommissionTrigger
+
+main : [Daml.Trigger.Trigger ()]
```
Comment (tamaskalcza-da) [inline triggers/src/main/daml/DA/RefApps/Bond/Triggers/TriggersLibrary.daml:17]: Is this needed? If not, remove.

File: ui-js/src/participants.json [hunk 3]
```diff
@@ -0,0 +1,71 @@
+{
```
Comment (tamaskalcza-da) [inline ui-js/src/participants.json:1]: Is this different from the other `participants.json`?

**Review by tamaskalcza-da — CHANGES_REQUESTED** [review]

File: ui-js/src/participants.json (hunk 3 above)
Comment (gaborh-da) [reply ui-js/src/participants.json:1]: No, I think this is the same.
See the following part in the README:
https://github.com/digital-asset/ex-bond-issuance/pull/163/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R131

Comment (gaborh-da) [reply ui-js/src/participants.json:1]: In the config, there is this part however:
https://github.com/digital-asset/ex-bond-issuance/pull/163/files#diff-403776bd79ad1c80b3e02fa443bf3fd9d3bdffa9315da03a967fe66041c798bcR7

So we need (unfortunately) the `participants.json` to be present in the ui dir "at UI compilation time".

I know this is not the best, but did not have an idea how to solve this.
Note that the file is only used in "non local dev mode".

I know that this is the same thing we discuss all the time how one needs to handle configuration in UI code but I still do not know the best solution for our case. We would need to investigate this and reshape all the refapps including this (but that is a different task, different PR).

File: README.md [hunk 4]
```diff
@@ -78,6 +78,59 @@ Reset the application by following these steps:
...
+
+AuctionAgent:
+DA.RefApps.Bond.Triggers.AuctionFinalizeTrigger:auctionFinalizeTrigger
+
+CSD:
+DA.RefApps.Bond.Triggers.RedemptionCalculationTrigger:redemptionCalculationTrigger
+
+```
+
+6. Run `make buildui`. Copy `participants.json` into `ui-js/src` with `cp participants.json ui-js/src/`. Run `npm run build` in `ui-js`. Then run `zip -r bondui.zip build/`. Upload `bondui.zip` to DABL to deploy the UI.
```
Comment (richardkapolnai-da) [inline README.md:131]: Feel free to ignore: it's more elegant to expose this feature in the Makefile.
(A close example, but this part is unfinished: https://github.com/digital-asset/ex-supply-chain/blob/ERA-847-ui/Makefile)

File: src/main/daml/DA/RefApps/Bond/Test/MarketSetupScript.daml (hunk 1 above)
Comment (chrisnorris-da) [reply src/main/daml/DA/RefApps/Bond/Test/MarketSetupScript.daml:49]: That would take operator out of scope .. so not the same.

**Review by tamaskalcza-da — APPROVED** [review]

**Comment by gaborh-da** [comment]
I checked once again and it works with the merged master.

**Pull request merged**